pkg/controller/cincinnati/cincinnati_controller_test.go

```diff
@@ -52,17 +57,47 @@ func TestMain(m *testing.M) {
 
 // TestReconcileComplete ensures that the cincinnati object is reconciled successfully
 func TestReconcileComplete(t *testing.T) {
-	cincinnati := newDefaultCincinnati()
-	r := newTestReconciler(cincinnati)
-	request := newRequest(cincinnati)
+	tests := []struct {
+		name                   string
+		cincinnati             *cv1alpha1.Cincinnati
+		expectedConflictStatus corev1.ConditionStatus
+	}{
+		{
+			name:                   "Reconcile",
+			cincinnati:             newDefaultCincinnati(),
+			expectedConflictStatus: corev1.ConditionFalse,
+		},
+		{
+			name: "ReconcileWithGraphDataImage",
+			cincinnati: func() *cv1alpha1.Cincinnati {
+				cincinnati := newDefaultCincinnati()
+				cincinnati.Spec.GraphDataImage = testGraphDataImage
+				return cincinnati
+			}(),
+			expectedConflictStatus: corev1.ConditionTrue,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			cincinnati := test.cincinnati
+			r := newTestReconciler(cincinnati)
+			request := newRequest(cincinnati)
 
-	_, err := r.Reconcile(request)
-	if err != nil {
-		t.Fatal(err)
+			_, err := r.Reconcile(request)
+			if err != nil {
+				t.Fatal(err)
+			}
+			instance := &cv1alpha1.Cincinnati{}
+			err = r.client.Get(context.TODO(), request.NamespacedName, instance)
+			if err != nil {
+				t.Fatal(err)
+			}
+			assert.True(t, conditionsv1.IsStatusConditionTrue(instance.Status.Conditions, cv1alpha1.ConditionReconcileCompleted))
+
+			assert.Equal(t, instance.Status.Conditions[1].Type, cv1alpha1.ConditionConfigurationConflict)
```

Comment:
Instead of checking multiple status conditions in different ways, how about using `expectedConditions []conditionsv1.Condition` in the struct definition, set the required `Type` and `Status` in the test case and then loop through it to check for conditions,
```
	for _, condition := range test.expectedConditions {
		assert.True(t, conditionsv1.IsStatusConditionPresentAndEqual(instance.Status.Conditions, condition.Type, condition.Status))
	}
	if instance != nil {
		assert.Equal(t, len(test.expectedConditions), len(test.instance.Status.Conditions))
	}
```
This then can be added in a utility function and other testcases can use it as well.

Reply:
Will do it if there is a bit more repetition. Right now, it's just 2 lines. :smile:

Reply:
It's odd that we are relying on `instance.Status.Conditoins[1].Type` to be the `ConditionConfigurationConflict`...which means that we know another condition should be defined that we aren't checking. My belief is that we should be explicit in what we want the conditions to be after calling the function under test (in this case `Reconcile`) and verify that those conditions are there (and no others).

Reply:
Fair enough. Added expectedConditions to the struct.